test/tests/PlayTests.cpp

```diff
@@ -0,0 +1,139 @@
+#include <openrct2/world/MapAnimation.h>
+#include <openrct2/world/Park.h>
+#include <openrct2/world/Scenery.h>
+#include <openrct2/world/Sprite.h>
+#include <string>
+
+using namespace OpenRCT2;
+
+class PlayTests : public testing::Test
+{
+};
+
+static std::unique_ptr<IContext> localStartGame(const std::string& parkPath)
+{
+    gOpenRCT2Headless = true;
+    gOpenRCT2NoGraphics = true;
+    core_init();
+
+    auto context = CreateContext();
+    if (!context->Initialise())
+        return {};
+
+    auto importer = ParkImporter::CreateS6(context->GetObjectRepository());
+    auto loadResult = importer->LoadSavedGame(parkPath.c_str(), false);
+    context->GetObjectManager().LoadObjects(loadResult.RequiredObjects.data(), loadResult.RequiredObjects.size());
+    importer->Import();
+
+    reset_sprite_spatial_index();
+
+    reset_all_sprite_quadrant_placements();
+    scenery_set_default_placement_configuration();
+    load_palette();
+    map_reorganise_elements();
+    sprite_position_tween_reset();
+    AutoCreateMapAnimations();
+    fix_invalid_vehicle_sprite_sizes();
+
+    gGameSpeed = 1;
+
+    return context;
+}
+
+template<class Fn> static bool updateUntil(GameState& gs, int maxSteps, Fn&& fn)
+{
+    while (maxSteps-- && !fn())
+    {
+        gs.UpdateLogic();
+    }
+    return maxSteps > 0;
+}
+
+TEST_F(PlayTests, SecondGuestInQueueShouldNotRideIfNoFunds)
+{
+    /* This test verifies that a guest, when second in queue, won't be forced to enter
+     * the ride if it has not enough money to pay for it.
+     * To simulate this scenario, two guests (a rich and a poor) are encouraged to enter
+     * the ride queue, and then the price is raised such that the second guest in line
+     * (the poor one) cannot pay. The poor guest should not enter the ride.
+     */
+    std::string initStateFile = TestData::GetParkPath("small_park_with_ferris_wheel.sv6");
+
+    auto context = localStartGame(initStateFile);
+    ASSERT_NE(context.get(), nullptr);
+
+    auto gs = context->GetGameState();
+    ASSERT_NE(gs, nullptr);
+
+    // Open park for free but charging for rides
+    GameActions::Execute(std::make_unique<ParkSetParameterAction>(ParkParameter::Open).get());
+    park_set_entrance_fee(0);
+    gParkFlags |= PARK_FLAGS_UNLOCK_ALL_PRICES;
+
+    // Find ferris wheel
+    auto rideManager = GetRideManager();
+    auto it = std::find_if(
+        rideManager.begin(), rideManager.end(), [](auto& ride) { return ride.type == RIDE_TYPE_FERRIS_WHEEL; });
+    ASSERT_NE(it, rideManager.end());
+    Ride& ferrisWheel = *it;
+
+    // Open it for free
+    ride_set_status(&ferrisWheel, RIDE_STATUS_OPEN);
+    GameActions::Execute(std::make_unique<RideSetPriceAction>(ferrisWheel.id, 0, true).get());
+
+    // Ignore intesity to stimulate peeps to queue into ferris wheel
+    gCheatsIgnoreRideIntensity = true;
+
+    // Insert a rich guest
+    auto richGuest = gs->GetPark().GenerateGuest();
+    richGuest->cash_in_pocket = 3000;
```

Comment:
Really rich for `OpenRCT2` standards :D

Reply:
That value actuallly means 300.0 in the game. that should be fine, no?

Reply:
Sure, I was just poking at you 🤣